src/details/ArborX_DetailsTreeConstruction.hpp

```diff
@@ -36,6 +36,19 @@ namespace ArborX
 {
 namespace Details
 {
+
+template <typename T, typename TTag>
+struct TagHelper
+{
+private:
+  using accessor_return_type =
+      std::decay_t<decltype(Traits::Access<T, TTag>::get(
+          std::declval<T const &>(), std::declval<int>()))>;
```

Comment:
Why do you need the second `declval`?

Reply:
Could probably replace with an integer but the second argument is needed because `get` takes two params.